### ISSUE TITLE
Custom root SSL/TLS certificate is installed, but it doesn't work on GrapheneOS

### ISSUE THREAD

**Issue by s0llvan** [GrapheneOS/os-issue-tracker#4412]
## Description

I generated and installed a custom root certificate for my local network using the following commands and configuration files :

### Configuration Files

**root.cnf**
```
[ req ]
prompt             = no
string_mask        = default
default_bits       = 2048
distinguished_name = req_distinguished_name
x509_extensions    = x509_ext

[ req_distinguished_name ]
countryName = fr
organizationName = LocalNetwork
commonName = LocalNetwork Root CA

[ x509_ext ]
basicConstraints=critical,CA:true,pathlen:0
keyUsage=critical,keyCertSign,cRLSign
```

**server.cnf**
```
[ req ]
prompt             = no
string_mask        = default
default_bits       = 2048
distinguished_name = req_distinguished_name
x509_extensions    = x509_ext

[ req_distinguished_name ]
countryName = fr
organizationName = LocalNetwork
commonName = Certificate Applications LocalNetwork

[ x509_ext ]
keyUsage=critical,digitalSignature,keyAgreement
subjectAltName = @alt_names

[ alt_names ]
DNS.1 = mywebserver.localnetwork
```

## Commands used

```
openssl req -x509 -new -days 3650 -keyout localnetwork.key -out localnetwork.cer -config root.cnf -extensions x509_ext
openssl req -nodes -new -keyout server.key -out server.csr -config server.cnf -extensions x509_ext
openssl x509 -req -in server.csr -CA localnetwork.cer -CAkey localnetwork.key -set_serial 123 -out server.crt -extfile server.cnf -extensions x509_ext
openssl x509 -outform PEM -in localnetwork.cer -out localnetwork.crt
```

## Problem

The generated root certificate (`localnetwork.crt`) was correctly installed on a device running GrapheneOS using the CA certificate management option. The certificate was added to both **Trusted Credentials** and **User Credentials**. I also tested the installation using both the `.crt` and `.cer` file formats.  

However:  

- **On GrapheneOS:** Secure connections using this certificate (e.g., to `https://mywebserver.localnetwork`) are not recognized. A certificate error appears in the browser, and applications relying on HTTPS fail to connect.  
- **On other platforms:** The certificate works as expected. It is recognized, and secure connections are established successfully on other Android devices (non-GrapheneOS) and Linux systems (e.g., Ubuntu).

## Environment

- **Device:** Pixel 8  
- **OS:** Android 15  
- **Build Number:** 2024112700

**Comment by s0llvan** [comment]
> Only apps opting into using user installed certificates will use non-system certificates on Android. It works the same as it usually does.

Yes, but Firefox does this, and it doesn't work on GrapheneOS, but it works on other ROMs

**Comment by s0llvan** [comment]
The certificate is not expired and works on the standard Android OS, that's why I am asking the question.

*OS, my mistake.

**Comment by s0llvan** [comment]
Yes, I’m talking about stock operating systems, but not necessarily on a Pixel with Android 15, this also includes those from other manufacturers.
What I have trouble understanding is why certificates work on stock systems but not on systems like GrapheneOS, even though it’s still possible to install them.

**Comment by s0llvan** [comment]
@thestinger I don't understand, why did you close the issue...? I still have the same problem only on GrapheneOS and not on my other Android devices... can't it be reopened ?
What's the point of being able to install certificates if they are unusable on GrapheneOS ?

**Comment by s0llvan** [comment]
Do you have an example of a certificate that works on GrapheneOS, please ?
Because I've also tested other certificates, but they don't work either.

Why would it be necessary to generate a different certificate for GrapheneOS compared to other operating systems ?

**Comment by s0llvan** [comment]
> The feature works on GrapheneOS. There's something wrong with the certificate you're trying to use.

Do you have an example of a certificate that works on GrapheneOS, please ?
Because I've also tested other certificates, but they don't work either.

Why would it be necessary to generate a different certificate for GrapheneOS compared to other operating systems ?

**Comment by s0llvan** [comment]
Alright, thanks

**Comment by martijnboers** [comment]
Having the exact same issues, working root.crt on my Android 15 Pixel, moved to GOS and can no longer use this certificate. It's a showstopper as I host a ton of internal tools that I have no interest in having on the web so I can use a pre-installed system CA. 

Really scratching my head on how to continue

**Comment by martijnboers** [comment]
> @martijnboers The exact same certificate worked on the stock OS?

Yes, I even regenerated a new one but same behavior. So now I'm limited to apps that use user certs or disable checking entirely

**Comment by s0llvan** [comment]
So apparently, I was right, there is indeed a bug @thestinger  but it seems they prefer to close the issue directly rather than look for a solution 🙄

**Comment by s0llvan** [comment]
@thestinger 

So apparently, I was right, that's exactly what I was saying, there is indeed a bug, but it seems they prefer to close the issue directly rather than look for a solution 😐🙄
